### PR TITLE
Annotate `AtomicReference` for nullness.

### PR DESCRIPTION
**Pull request by cpovirk**
Compare
https://github.com/jspecify/jdk/commit/a398fe4009ddf77578af49697c3f3a4211fc8adb

See also some discussion in https://github.com/jspecify/jdk/pull/27

### CODE REVIEW

**Comment by cpovirk** [comment]
Obviously this PR follows the `ThreadLocal` precedent with `class AtomicReference<@Nullable T>`. I'm not married to that, and in fact I expect that probably override it back to `AtomicReference<T>` inside Google :) So let me know what you think.

File: src/java.base/share/classes/java/util/concurrent/atomic/AtomicReference.java [hunk 1]
```diff
@@ -37,6 +37,7 @@
 
 import org.checkerframework.checker.interning.qual.UsesObjectEquals;
```
Comment (wmdietl) [inline src/java.base/share/classes/java/util/concurrent/atomic/AtomicReference.java:38]:
```suggestion
import org.checkerframework.checker.interning.qual.UsesObjectEquals;
import org.checkerframework.checker.nullness.qual.Nullable;
```

**Review by wmdietl — APPROVED** [review]
Looks good.

**Comment by wmdietl** [comment]
I made https://github.com/eisop/checker-framework/pull/563 , which should fix the downstream failures.

**Pull request merged**